codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydafny/DafnyApiCodegen.java

```diff
@@ -911,6 +912,43 @@ private TokenTree OperationMemberEnsures(
       .lineSeparated();
   }
 
+  private TokenTree MutableResourcesOnlyAddFreshObjects(
+    OperationShape operationShape
+  ) {
+      return operationShape
+        .getInput()
+        .map(shapeId -> model.expectShape(shapeId, StructureShape.class))
```

Comment:
Issue: I am concerned about what happens with the Unit shape.
(Which I am fairly certain is a structure.)

The filters here would just pull it out and it would be ignored, right?